proof/refine/AARCH64/Tcb_R.thy

```diff
@@ -1655,6 +1659,51 @@ lemma setSchedulerAction_invs'[wp]:
 
 end
 
+lemma setFlags_corres[corres]:
+  "flags = word_to_tcb_flags flags' \<Longrightarrow>
+   corres dc (tcb_at t and pspace_aligned and pspace_distinct) \<top>
+         (set_flags t flags) (setFlags t flags')"
+  unfolding set_flags_def setFlags_def
+  apply (corres corres: threadset_corres)
+  by (clarsimp simp: tcb_relation_def inQ_def)+
+
+crunch set_flags
+  for pspace_aligned[wp]: pspace_aligned
+  and pspace_distinct[wp]: pspace_distinct
+  and valid_cur_fpu[wp]: valid_cur_fpu
+
+context begin interpretation Arch . (*FIXME: arch-split*)
+
+lemma tcbFlagToWord_bit:
+  "\<exists>n. tcbFlagToWord flag = bit n"
+  by (auto simp: tcbFlagToWord_def archTcbFlagToWord_def split: tcb_flag.splits arch_tcb_flag.splits
+       simp del: bit_def)
+
+lemma word_to_tcb_flags_subtract:
+  "word_to_tcb_flags (flags && ~~ flags') = word_to_tcb_flags flags - word_to_tcb_flags flags'"
```

Comment:
nice that these work out

Reply:
I remember these being the usual word fun to prove, but yeah, very nice that it ended up working smoothly.